src/containers/SingleProjectPageContainer/index.js

```diff
@@ -11,22 +11,43 @@ type Props = {
   projectSlug: string
 }
 
-const SingleProjectPageContainer = (props: Props) => (
-  <Wrapper flex contentCentered>
-    <Query query={query} variables={{ projectSlug: props.projectSlug }}>
-      {({ error, loading, data }) => {
-        if (error) {
-          return <ErrorPage statusCode={404} />
-        }
-
-        if (loading) {
-          return <Loader isCentered withText isDark />
-        }
-
-        return <View />
-      }}
-    </Query>
-  </Wrapper>
-)
+class SingleProjectPageContainer extends React.Component<Props> {
+  static getInitialProps(context: any) {
+    const { projectSlug } = context.query
+
+    // We can return an actual 404 before rendering the <ErrorPage> server-side.
+    if (!projectSlug && context.res) {
+      context.res.statusCode = 404
+    }
+
+    return { projectSlug }
+  }
+
+  render() {
+    const { projectSlug } = this.props
+
+    if (!projectSlug) {
+      return <ErrorPage statusCode={404} />
+    }
+
+    return (
+      <Wrapper flex contentCentered stretch>
+        <Query query={query} variables={{ projectSlug }}>
+          {({ error, loading, data }) => {
+            if (error) {
+              return <ErrorPage statusCode={404} />
```

Comment:
The only error should be that the project doesn’t exist. Let’s not handle it, yolo. :)